lib/resize.js

```diff
@@ -2,9 +2,21 @@ const Jimp = require('jimp');
 const TinyColor = require('tinycolor2');
 
 module.exports = async (options, buffer) => {
-  const { width, height, background } = options;
+  const { background, type } = options;
+  let { width, height } = options;
+  // if only one of width/height was provided, make it a square:
+  if (width && !height) {
```

Comment:
if you only provide one, then it should auto decide the other dimension - that should be part of jimp, right?

Reply:
You would think, but nope it only works if you provide both.

Reply:
https://github.com/oliver-moran/jimp/tree/master/packages/plugin-resize#usage - if it doesn't exist, set to auto

Reply:
https://github.com/oliver-moran/jimp/tree/master/packages/plugin-resize#usage - if one or the other is not provided, set to auto

Reply:
This works for resize, but cover and contain still fail if you don't provide both.  This covers
all possible cases.